### ISSUE TITLE
ENV replacement doesn't account for `globalThis.process.env`

### ISSUE THREAD

**Issue by johnrees** [farm-fe/farm#982]
### Steps to reproduce

Add `import * as graphql from "graphql"; console.log(graphql);` to a file and try to build the project

### What is expected?

the project should build, the compiled output of [this line](https://github.com/graphql/graphql-js/blob/main/src/jsutils/instanceOf.ts#L12) should be the equivalent of `"production" === 'production'`

### What is actually happening?

The output is `globalThis.process && globalThis."production" === 'production'`

```
(node:10950) ExperimentalWarning: Import assertions are not a stable feature of the JavaScript language. Avoid relying on their current behavior and syntax as those might change in a future version of Node.js.
(Use `node --trace-warnings ...` to show where the warning was created)
[ Farm ] Using config file at /Users/john/Code/farm-project/farm.config.ts
[ Farm ] Error: Parse `node_modules/.pnpm/graphql@16.8.1/node_modules/graphql/jsutils/instanceOf.mjs` failed.
 Error:
  × Expected ident
    ╭─[node_modules/.pnpm/graphql@16.8.1/node_modules/graphql/jsutils/instanceOf.mjs:9:1]
  9 │ export const instanceOf =
 10 │   /* c8 ignore next 6 */
 11 │   // FIXME: https://github.com/graphql/graphql-js/issues/2317
 12 │   globalThis.process && globalThis."production" === 'production'
    ·                                    ────────────
 13 │     ? function instanceOf(value, constructor) {
 14 │         return value instanceof constructor;
 15 │       }
    ╰────

Potential Causes:
1.The module have syntax error.
2.This kind of module is not supported, you may need plugins to support it

 ELIFECYCLE  Command failed with exit code 1.
 ```

### System Info

```Shell
System:
    OS: macOS 14.2.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 2.28 GB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.18.2 - ~/Library/Caches/fnm_multishells/5714_1707992194832/bin/node
    Yarn: 4.0.2 - ~/Library/Caches/fnm_multishells/5714_1707992194832/bin/yarn
    npm: 10.2.4 - ~/Library/Caches/fnm_multishells/5714_1707992194832/bin/npm
    pnpm: 8.15.3 - ~/Library/Caches/fnm_multishells/5714_1707992194832/bin/pnpm
    bun: 1.0.26 - ~/.bun/bin/bun
  Browsers:
    Chrome: 117.0.5938.149
    Safari: 17.2.1
```

### Any additional comments?

- related vite issue https://github.com/vitejs/vite/issues/12185
- I found another package that was using `global.process.env` which had the same problem

**Comment by johnrees** [comment]
I've fixed it with this for now

```
 plugins: [
    {
      name: "replace-process-env",
      load: {
        filters: {
          resolvedPaths: ["\\/instanceOf\\.mjs$"],
        },
        executor: (params, context) => {
          const { resolvedPath } = params;
          const content = readFileSync(resolvedPath, "utf-8");
          return {
            content: content.replace("globalThis.process.env", "process.env"),
            moduleType: "js",
          };
        },
      },
    },
  ],
  ```

**Comment by wre232114** [comment]
I'll take a look. thanks for reporting issues.

**Comment by wre232114** [comment]
fixed in `@farmfe/core@0.16.9`
